dolphinscheduler-task-plugin/dolphinscheduler-task-flink/src/main/java/org/apache/dolphinscheduler/plugin/task/flink/FlinkConstants.java

```diff
@@ -24,19 +24,48 @@ private FlinkConstants() {
     }
 
     /**
-     * flink
+     * flink command
+     * usage: flink run [OPTIONS] <jar-file> <arguments>
+     */
+    public static final String FLINK_COMMAND = "flink";
+    public static final String FLINK_RUN = "run";
+
+    /**
+     * flink sql command
+     * usage: sql-client.sh -i <initialization file>, -j <JAR file>, -f <script file>...
+     */
+    public static final String FLINK_SQL_COMMAND = "sql-client.sh";
```

Comment:
Please note that flink ```sql-client.sh -f``` is only supported after version 1.13. And ```sql-client.sh``` at version 1.13 and 1.14. and 1.15 have some completely incompatible contents in these three versions.  If we add flink sql support. We should put it in the frontend page and document to let users distinguish it.

Reply:
> Please note that flink `sql-client.sh -f` is only supported after version 1.13. And `sql-client.sh` at version 1.13 and 1.14. and 1.15 have some completely incompatible contents in these three versions. If we add flink sql support. We should put it in the frontend page and document to let users distinguish it.
Flink sql requires sql-client.sh -f <script file> to facilitate users to edit sql scripts. I set it on the front end. When the user selects the flink task and selects sql, the version of flink is >= 1.13 and cannot be changed. flink1.15 also supports sql-client.sh -f https://nightlies.apache.org/flink/flink-docs-release-1.15/zh/docs/dev/table/sqlclient/

Reply:
Flink sql requires sql-client.sh -f <script file> to facilitate users to edit sql scripts. I set it on the front end. When the user selects the flink task and selects sql, the version of flink is >= 1.13 and cannot be changed. flink1.15 also supports sql-client.sh -f 
【flink 1.15 sql client】https://nightlies.apache.org/flink/flink-docs-release-1.15/zh/docs/dev/table/sqlclient/
【flink 1.16 sql client】https://nightlies.apache.org/flink/flink-docs-release-1.16/zh/docs/dev/table/sqlclient/

Reply:
Yes. I think we should also add the new features of Flink 1.13 to flink task. For example ```-i``` from UI or resource center and ```--jar``` from resource center. In particular, I think the ```-i``` option will be used in most scenarios when users use multiple connectors.

Reply:
Users want us to support flink sql as soon as possible. When the user enters the sql script in the editor, we roll it back to a file and execute this file with flink `sql-client.sh -f`. Features like `-i` or `--jar` are not included for now. Supplements can be added later if desired

Reply:
```-jar``` supplement can be added later is ok to me. However, almost 90% of the Flink SQL related operations need to be performed based on ```-i```. I think it's needed in flink sql feature.

Reply:
【-i】: Script file that used to init the session context. If get error in execution, the sql client will exit.Notice it's not allowed to add query or insert into the init file. 

The initialization script specified by -i is used by most users to specify some parameters like set a=b. Of course, these parameters can also be placed in the SQL script specified by -f. So our users also want to set the parameters of flink themselves in the front-end editor. Of course, we also do built-in processing for the flink parameters filled in by the user front-end in the FlinkTask class. These parameters are as follows:

```sql
set sql-client.execution.result-mode=tableau;
set execution.target=yarn-per-job;
set yarn.application.name=streaming-top-speed-windowing-sql;
set yarn.application.queue=default;
set jobmanager.memory.process.size=1024mb;
set taskmanager.memory.process.size=4096mb;
set taskmanager.numberOfTaskSlots=2;
set parallelism.default=4;
set execution.runtime-mode=streaming;
```

At the same time -i can be specified directly under the control of the front-end option parameter ～

Reply:
> 【-i】: Script file that used to init the session context. If get error in execution, the sql client will exit.Notice it's not allowed to add query or insert into the init file.
> 
> The initialization script specified by -i is used by most users to specify some parameters like set a=b. Of course, these parameters can also be placed in the SQL script specified by -f. So our users also want to set the parameters of flink themselves in the front-end editor. Of course, we also do built-in processing for the flink parameters filled in by the user front-end in the FlinkTask class. These parameters are as follows:
> 
> ```sql
> set sql-client.execution.result-mode=tableau;
> set execution.target=yarn-per-job;
> set yarn.application.name=streaming-top-speed-windowing-sql;
> set yarn.application.queue=default;
> set jobmanager.memory.process.size=1024mb;
> set taskmanager.memory.process.size=4096mb;
> set taskmanager.numberOfTaskSlots=2;
> set parallelism.default=4;
> set execution.runtime-mode=streaming;
> ```
> 
> At the same time -i can be specified directly under the control of the front-end option parameter ～

I do not think so. I think the initialization script specified by ```-i``` is used by most users to using different connectors. Users will execute ```DDL``` or ```USE CATALOG``` scripts in order to create a ```flink-cdc``` or ```kafka``` table and use it.

Reply:
I get it! I think there are two ways：
（1）Edit the initialization script in [**resource center**], select this resource when creating a task, and then specify -i directly in [**pption parameters**]. This way I just took off this pr to verify that it works.
（2）The front end provides **two** editors, one for filling in the user's initialization script and one for filling in FlinkSQL, which may be a little clearer.
Could you help me with a better solution? We'll just optimize part of code immediately, Grazie！

Reply:
For the first solution above, the current pr already satisfies the function of `-i`, do you think you need to continue to modify it? 

Of course I'm more than willing to optimize the second solution, and then I can continue to optimize in the next pr.

 It's worth noting that documentation is being added, please wait.

Reply:
> For the first solution above, the current pr already satisfies the function of `-i`, do you think you need to continue to modify it?
> 
> Of course I'm more than willing to optimize the second solution, and then I can continue to optimize in the next pr.
> 
> It's worth noting that documentation is being added, please wait.

It would be friendly if you could enlight me where you implement ```-i``` to enable users to specify it.

Reply:
The front end provides two editors, one for filling in the user's initialization script and the other for filling in FlinkSQL.

 then I will continue to optimize this pr according to this plan